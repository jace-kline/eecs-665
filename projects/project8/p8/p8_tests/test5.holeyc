int a;
int b;

int f(int x) {
    x = 7;
    x++;
    x++;
    return x;
}

int main() {
    a = 5;
    int b;
    b = f(a);
    b = a;
    return 0;
}