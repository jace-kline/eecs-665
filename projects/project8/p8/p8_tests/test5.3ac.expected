[BEGIN GLOBALS]
a
b
[END GLOBALS]
[BEGIN f LOCALS]
x (formal)
[END f LOCALS]
fun_f:      enter f
            getarg 1 [x]
            [x] := 7
            [x] := 8
            [x] := 9
            setret 9
lbl_0:      leave f
[BEGIN main LOCALS]
b (local)
tmp0 (tmp)
[END main LOCALS]
main:       enter main
            [a] := 5
            setarg 1 5
            call f
            getret [tmp0]
            [b] := [tmp0]
            [b] := [a]
            setret 0
lbl_1:      leave main
